contrib/pg_tde/src/access/pg_tde_tdemap.c

```diff
@@ -209,38 +215,42 @@ tde_sprint_key(InternalKey *k)
 }
 
 /*
- * Generates a new internal key for WAL and adds it to the _dat file. It doesn't
+ * Generates a new internal key for WAL and adds it to the _map file. It doesn't
  * add unecnrypted key into cache but rather sets it in `rel_key_data`.
  *
  * We have a special function for WAL as it is being called during recovery
  * (start) so there should be no XLog records, aquired locks, and reads from
  * cache. The key is always created with start_lsn = InvalidXLogRecPtr. Which
  * will be updated with the actual lsn by the first WAL write.
  */
-void
-pg_tde_create_wal_key(InternalKey *rel_key_data, const RelFileLocator *newrlocator, uint32 entry_type)
+InternalKey *
+pg_tde_create_wal_key(const RelFileLocator *newrlocator, uint32 entry_type)
 {
 	TDEPrincipalKey *principal_key;
+	RelKeyCacheRec *rec;
 
 	LWLockAcquire(tde_lwlock_enc_keys(), LW_SHARED);
 
 	principal_key = GetPrincipalKey(newrlocator->dbOid, LW_SHARED);
 	if (principal_key == NULL)
 	{
 		ereport(ERROR,
-				(errmsg("principal key not configured"),
+				(errmsg("principal key is not configured"),
```

Comment:
If this should be fixed it should be fixed in all places and as a separate commit.